### PR TITLE
docs: add lightdash yml schema spec

### PR DESCRIPTION
**Pull request by owlas**
### Pre requirements:
- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: 

### Scope of the changes (select all that apply):
- [ ] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

File: packages/common/src/schemas/json/lightdash-0.1.json [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                                                    "enum": [
+                                                        "km",
+                                                        "mi",
+                                                        "usd",
+                                                        "gbp",
+                                                        "eur",
+                                                        "percent"
+                                                    ]
+                                                },
+                                                "time_intervals": {
```
Comment (ZeRego) [inline packages/common/src/schemas/json/lightdash-0.1.json:165]: We are missing a few possible values here. You can check the docs on all possible values https://docs.lightdash.com/references/dimensions#dimension-configuration

File: packages/common/src/schemas/json/lightdash-0.1.json [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                                        "type": "string"
+                                    }
+                                },
+                                "meta": {
+                                    "type": "object",
+                                    "properties": {
+                                        "metrics": {
+                                            "type": "object",
+                                            "patternProperties": {
+                                                "^[a-z0-9_]+$": {
```
Comment (ZeRego) [inline packages/common/src/schemas/json/lightdash-0.1.json:85]: Should we support uppercase letters ?

File: packages/common/src/schemas/json/lightdash-0.1.json [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+                "required": ["name"]
```
Comment (rephus) [inline packages/common/src/schemas/json/lightdash-0.1.json:186]: is description not required ?

Comment (owlas) [reply packages/common/src/schemas/json/lightdash-0.1.json]: It's not!

File: packages/common/src/schemas/json/lightdash-0.1.json (hunk 2 above)
Comment (owlas) [reply packages/common/src/schemas/json/lightdash-0.1.json]: do we support uppercase letters for field names? @ZeRego

Comment (owlas) [reply packages/common/src/schemas/json/lightdash-0.1.json]: decided to leave this as lower case only

File: packages/common/src/schemas/json/lightdash-0.1.json (hunk 1 above)
Comment (owlas) [reply packages/common/src/schemas/json/lightdash-0.1.json]: Thanks!

**Pull request merged**